actionpack/lib/action_dispatch/http/mime_type.rb

```diff
@@ -18,15 +20,16 @@ def each
 
     def <<(type)
       @mimes << type
-      @symbols = nil
+      @symbols << type.to_sym
     end
 
     def delete_if
-      @mimes.delete_if { |x| yield x }.tap { @symbols = nil }
-    end
-
-    def symbols
-      @symbols ||= map(&:to_sym)
```

Comment:
No need for this memoization, `to_sym` doesn't do anything expensive, it just returns a instance variable https://github.com/rails/rails/blob/785427b88c01654a441b8a7ed602feb35358285b/actionpack/lib/action_dispatch/http/mime_type.rb#L250-L252